File: interview_questions/python/word_search_test.py
from word_search import word_search


def test_horizontal_word():
    assert word_search("dog", [["d", "o", "g"], ["d", "f", "s"]]) is True


def test_vertical_word():
    assert (
        word_search("dog", [["d", "r", "r"], ["o", "f", "s"], ["g", "d", "s"]]) is True
    )


def test_no_word():
    assert (
        word_search("dog", [["d", "r", "r"], ["b", "f", "s"], ["g", "d", "s"]]) is False
    )

File: interview_questions/ocaml/word_search_test.ml

open OUnit2
open Word_search


let tests = "test suite for searching words" >::: [
    "word is there vertically and horizontally"  >:: (fun _ -> assert_equal (ref true) (word_search "dfsa" [|
        [| 'd';'o';'b' |];
        [| 'f';'a';'t' |];
        [| 's';'a';'t' |]
                                                                      |]));
    "word is not there"  >:: (fun _ -> assert_equal (ref false) (word_search "dog" [|
        [| 'd';'o';'b' |];
        [| 'f';'a';'t' |]
                                                                      |]));
    "word exists"  >:: (fun _ -> assert_equal (ref true) (word_search "dog" [|
        [| 'd';'o';'g' |];
        [| 'f';'a';'t' |]
                                                                      |]));
]


let _ = run_test_tt_main tests
